sources/kms/alias.go

```diff
@@ -0,0 +1,127 @@
+				ErrorString: "aliasName is nil",
+			}
+		}
+
+		// Ignore AWS managed keys, they are predefined and might not have a target key ID
+		if strings.HasPrefix(*alias.AliasName, "alias/aws/") {
+			// AWS managed keys
+			continue
+		}
+
+		// This should never happen except for AWS managed keys.
+		if alias.TargetKeyId == nil {
+			return nil, &sdp.QueryError{
+				ErrorType:   sdp.QueryError_OTHER,
+				ErrorString: "targetKeyId is nil",
+			}
+		}
+
+		// The uniqueAttributeValue for this is the combination of the keyID and aliasName
+		// i.e., "cf68415c-f4ae-48f2-87a7-3b52ce/alias/test-key"
+		err = attributes.Set("uniqueName", fmt.Sprintf("%s/%s", *alias.TargetKeyId, *alias.AliasName))
+		if err != nil {
+			return nil, err
+		}
+
+		item := sdp.Item{
+			Type:            "kms-alias",
+			UniqueAttribute: "uniqueName",
+			Attributes:      attributes,
+			Scope:           scope,
+		}
+
+		if alias.TargetKeyId == nil {
+			// +overmind:link kms-key
+			item.LinkedItemQueries = append(item.LinkedItemQueries, &sdp.LinkedItemQuery{
+				Query: &sdp.Query{
+					Type:   "kms-key",
+					Method: sdp.QueryMethod_GET,
+					Query:  *alias.TargetKeyId,
+					Scope:  scope,
+				},
+				BlastPropagation: &sdp.BlastPropagation{
+					// These are tightly linked
+					// Adding, deleting, or updating an alias can allow or deny permission to the KMS key.
+					In:  true,
+					Out: true,
+				},
+			})
+		}
+
+		items = append(items, &item)
+	}
+
+	return items, nil
+}
+
+//go:generate docgen ../../docs-data
+// +overmind:type kms-alias
+// +overmind:descriptiveType KMS Alias
+// +overmind:get Get an alias by keyID/aliasName
+// +overmind:list List all aliases
+// +overmind:search Search aliases by keyID
+// +overmind:group AWS
+// +overmind:terraform:queryMap aws_kms_alias.arn
+
+func NewAliasSource(client *kms.Client, accountID string, region string) *sources.DescribeOnlySource[*kms.ListAliasesInput, *kms.ListAliasesOutput, *kms.Client, *kms.Options] {
+	return &sources.DescribeOnlySource[*kms.ListAliasesInput, *kms.ListAliasesOutput, *kms.Client, *kms.Options]{
+		ItemType:  "kms-alias",
+		Client:    client,
+		AccountID: accountID,
+		Region:    region,
+		DescribeFunc: func(ctx context.Context, client *kms.Client, input *kms.ListAliasesInput) (*kms.ListAliasesOutput, error) {
+			return client.ListAliases(ctx, input)
+		},
+		InputMapperGet: func(_, query string) (*kms.ListAliasesInput, error) {
+			// query must be in the format of: the keyID/aliasName
+			// note that the aliasName will have a forward slash in it
+			// i.e., "cf68415c-f4ae-48f2-87a7-3b52ce/alias/test-key"
+			tmp := strings.Split(query, "/")
+			if len(tmp) < 2 {
+				return nil, &sdp.QueryError{
+					ErrorType:   sdp.QueryError_NOTFOUND,
+					ErrorString: fmt.Sprintf("query must be in the format of: the keyID/aliasName, but found: %s", query),
+				}
+			}
+
+			return &kms.ListAliasesInput{
+				KeyId: &tmp[0], // keyID
+			}, nil
+		},
+		UseListForGet: true,
```

Comment:
Nice use of the combination of `InputMapperGet` and `UseListForGet` here. Get request will be somewhat inefficient because we'll still have to list many aliases, but should be limited since we're at least only listing for the known key. Nice